pkg/alertmanager/multitenant.go

```diff
@@ -423,6 +429,8 @@ func (am *MultitenantAlertmanager) setConfig(cfg alerts.AlertConfigDesc) error {
 }
 
 func (am *MultitenantAlertmanager) newAlertmanager(userID string, amConfig *amconfig.Config) (*Alertmanager, error) {
+	reg := prometheus.NewRegistry()
+	am.metrics.addUserRegistry(userID, reg)
```

Comment:
Nit: I'd suggest adding registry to `am.metrics` only if creation of alert manager doesn't fail with error.

Reply:
👍